File: ios/flutter_native_ai/Sources/flutter_native_ai/FlutterNativeAiPlugin.swift
import Flutter

public class FlutterNativeAiPlugin: NSObject, FlutterPlugin {
  private let bridge = OnDeviceAiBridge()

  public static func register(with registrar: FlutterPluginRegistrar) {
    let instance = FlutterNativeAiPlugin()
    OnDeviceAiHostApiSetup.setUp(
      binaryMessenger: registrar.messenger(),
      api: instance.bridge
    )
    instance.bridge.registerStreamHandler(with: registrar.messenger())
  }
}

File: android/src/main/kotlin/com/bowvie/flutter_native_ai/FlutterNativeAiPlugin.kt
package com.bowvie.flutter_native_ai

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** FlutterNativeAiPlugin */
class FlutterNativeAiPlugin : FlutterPlugin {
    private var bridge: OnDeviceAiBridge? = null

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        val messenger = flutterPluginBinding.binaryMessenger
        bridge = OnDeviceAiBridge().also {
            OnDeviceAiHostApi.setUp(messenger, it)
            it.registerStreamHandler(messenger)
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        OnDeviceAiHostApi.setUp(binding.binaryMessenger, null)
        bridge?.close()
        bridge = null
    }
}
